### ISSUE TITLE
`cp.array` turns some array into `nan` ?

### ISSUE THREAD

**Issue by TakuNishiumi** [cupy/cupy#5320]
Hello!

I ran the below code and I got a strange result.
Do you know the reasons?
(I attached a zip file contains .npy file) 
[test_array_for_cupy.zip](https://github.com/cupy/cupy/files/6609874/test_array_for_cupy.zip)

```
import cupy as cp
import numpy as np
arr = np.load("./test_array_for_cupy.npy")
print("numpy",np.sum(np.isnan(np.array(arr)))) # returns 0
print("cupy", cp.sum(cp.isnan(cp.array(arr)))) # returns 415
```

Environment:
```
OS : Linux-4.15.0-143-generic-x86_64-with-debian-buster-sid
Python Version : 3.7.9
CuPy Version : 9.1.0
CuPy Platform : NVIDIA CUDA
NumPy Version : 1.20.3
SciPy Version : 1.6.3
Cython Build Version : 0.29.23
Cython Runtime Version : None
CUDA Root : /home/nishiumi/anaconda3/envs/py37
nvcc PATH : None
CUDA Build Version : 11020
CUDA Driver Version : 11020
CUDA Runtime Version : 11020
cuBLAS Version : 11401
cuFFT Version : 10401
cuRAND Version : 10203
cuSOLVER Version : (11, 1, 0)
cuSPARSE Version : 11401
NVRTC Version : (11, 2)
Thrust Version : 101000
CUB Build Version : 101000
Jitify Build Version :
cuDNN Build Version : 8100
cuDNN Version : 8001
NCCL Build Version : None
NCCL Runtime Version : None
cuTENSOR Version : None
cuSPARSELt Build Version : None
Device 0 Name : TITAN RTX
Device 0 Compute Capability : 75
Device 0 PCI Bus ID : 0000:1A:00.0
Device 1 Name : TITAN RTX
Device 1 Compute Capability : 75
Device 1 PCI Bus ID : 0000:68:00.0
```

**Comment by TakuNishiumi** [comment]
probably related to #3652

**Comment by kmaehashi** [comment]
Thanks for reporting, it's likely if your array is in big endian.

**Comment by kmaehashi** [comment]
Fixed via #5828
